src/quote/swap/adapters/zeroex.ts

```diff
@@ -0,0 +1,165 @@
+import axios, { AxiosRequestHeaders } from 'axios'
+
+import { ChainId } from 'constants/chains'
+import { SwapQuoteProvider, SwapQuoteRequest } from 'quote/swap/interfaces'
+
+type ZeroExApiSwapRequest = {
+  buyAmount?: string
+  buyToken: string
+  sellAmount?: string
+  sellToken: string
+  // 0xAPI expects percentage as value between 0-1 e.g. 5% -> 0.05
+  slippagePercentage?: string
+}
+
+export type ZeroExApiSwapResponseOrder = {
+  source: string
+  fillData: {
+    path?: string
+    pool?: {
+      tokens: string[]
+      poolAddress: string
+    }
+    tokenAddressPath?: string[]
+  }
+}
+
+export type ZeroExApiSwapResponseOrderBalancer = {
+  source: string
+  fillData: {
+    assets: string[]
+    chainId: number
+    swapSteps: {
+      poolId: string
+    }[]
+    vault: string
+  }
+}
+
+export type ZeroExApiSwapResponseOrderSushi = {
+  source: string
+  makerToken: string
+  takerToken: string
+  makerAmount: string
+  takerAmount: string
+  fillData: {
+    tokenAddressPath: string[]
+    router: string
+  }
+  fill: {
+    input: string
+    output: string
+    adjustedOutput: string
+    gas: number
+  }
+}
+
+export type ZeroExApiSwapResponse = {
+  buyAmount: string
+  buyTokenAddress: string
+  data: string
+  orders?:
+    | ZeroExApiSwapResponseOrder[]
+    | ZeroExApiSwapResponseOrderBalancer[]
+    | ZeroExApiSwapResponseOrderSushi[]
+  sellAmount: string
+  sellTokenAddress: string
+}
+
+export class ZeroExSwapQuoteProvider implements SwapQuoteProvider {
```

Comment:
Essentially, it's the same class as the `ZeroExApi` only adapted to the protocol.